libs/core/src/lib/utils/directives/carousel/carousel.directive.ts

```diff
@@ -0,0 +1,252 @@
+import {
+    AfterContentInit,
+    ChangeDetectorRef,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Input,
+    Output,
+    QueryList
+} from '@angular/core';
+import {
+    AnimationPlayer,
+} from '@angular/animations';
+import { CarouselItemDirective } from './carousel-item.directive';
+import * as Hammer from 'hammerjs';
+import { HammerConfig } from './carousel.module';
+import { Subject } from 'rxjs';
+
+
+export interface CarouselConfig {
+    vertical?: boolean;
+    elementsAtOnce?: number;
+    panSupport?: boolean
+    infinite?: boolean;
+    transition?: string;
+}
+
+export const Default_Transition_Duration: string = '150ms';
+
+
+@Directive({
+    selector: '[fdCarousel]',
+    host: {
+        class: 'fd-carousel'
+    }
+})
+export class CarouselDirective implements AfterContentInit {
+
+    /** Configuration for carousel */
+    @Input()
+    config: CarouselConfig;
+
+    /** Initial active item of carousel, position first + offset */
+    @Input()
+    active: CarouselItemDirective;
+
+    /** Whether support for mouse down, or tap should be enabled */
+    @Input()
+    panSupport: boolean = true;
```

Comment:
I don't understand the purpose of this input based on its name or description

Reply:
I changed it to `gestureSupport` and changed comment. If you have any proposition, feel free to put it there.